modules/apps/portal-store/portal-store-s3/src/main/java/com/liferay/portal/store/s3/S3FileCache.java

```diff
@@ -71,26 +71,36 @@ public class S3FileCache {
 	public void cleanUpCacheFiles() {
 		_calledCleanUpCacheFilesCount++;
 
-		if (_calledCleanUpCacheFilesCount <
-				_cacheDirCleanUpFrequency.intValue()) {
+		if ((_calledCleanUpCacheFilesCount <
```

Comment:
@mbowerman This counter is thread-unsafe, as it's a plain int. I would recommend to change it to `AtomicInteger` or `Semaphor`.